### PR TITLE
Google Tag Manager: Use Everywhere

### PR DESCRIPTION
**Pull request by danielcliu-ifixit**
## Overview

Previously we were using the GoogleTagManager only on the troubleshooting pages. Now, because we want to use Piwik Pro which we are loading via GTM, we want GTM too be loaded on basically every page. This moves the code that was used in the troubleshooting pages to its own file. It then also adds that code to every page.

## QA

Make sure the troubleshooting pages piwik still works, and make sure every other page is loading piwik too. Make sure that your `NEXT_PUBLIC_GTM_KEY` is set correctly.

Closes: #1990

### CODE REVIEW

File: frontend/templates/troubleshooting/index.tsx [hunk 1]
```diff
@@ -59,7 +59,8 @@ import { BreadCrumbs } from '@ifixit/breadcrumbs';
 import { HeadingSelfLink } from './components/HeadingSelfLink';
 import ProblemCard from './Problem';
 import { PixelPing } from '@components/analytics/PixelPing';
-import { TagManager, GoogleNoScript } from './components/TagManager';
+import { TagManager } from './components/TagManager';
```
[the remark this answers is not in the document]
Comment (danielcliu-ifixit) [reply frontend/templates/troubleshooting/index.tsx:62]: i dont think so. TagManager in this case is just a matomo wrapper, I guess i could have changed the name to be more clear but i didn't bother.

File: frontend/layouts/default/index.tsx [hunk 2]
```diff
@@ -323,6 +328,7 @@ const DefaultLayoutComponent = function ({
             </Box>
             <Matomo />
             <GoogleAnalytics />
+            <GoogleTagManger />
```
[the remark this answers is not in the document]
Comment (danielcliu-ifixit) [reply frontend/layouts/default/index.tsx:331]: Im not sure i follow. Why?

File: frontend/layouts/default/index.tsx [hunk 3]
```diff
@@ -323,6 +327,7 @@ const DefaultLayoutComponent = function ({
             </Box>
             <Matomo />
             <GoogleAnalytics />
+            <GoogleTagManager />
```
Comment (danielcliu-ifixit) [inline frontend/layouts/default/index.tsx:330]: We do not use the `GoogleNoScript()` in this file, and to be frank im not sure of its purpose. I think its for cases where javascript is disabled, but in that case wont most of the site be broken anyway? Do we want to track stats of people interacting with a broken version of our website?

File: frontend/layouts/default/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jarstelfox) [reply frontend/layouts/default/index.tsx]: We currently have two (duplicate) layout files. One for the next pages router, one for the next app router.

Any changes you make need to be copied between them to keep them in sync

**Review by jarstelfox — DISMISSED** [review]
CR 🌵

File: frontend/app/(defaultLayout)/components/GoogleTagManager.tsx
Comment (jarstelfox) [inline frontend/app/(defaultLayout)/components/GoogleTagManager.tsx:1]: Is this the same as the other `GoogleTagManager.tsx`? If so, then you should be fine to keep only one.

The only reason we forked was that their was API changes that made using the same code impossible with Matomo / GA (watching for hash changes mostly)

**Comment by danielcliu-ifixit** [comment]
weve decide to not use gtm anymore, and instead favor piwik

**Comment by jarstelfox** [comment]
> weve decide to not use gtm anymore, and instead favor loading piwik directly

Although this is true, I think adding GTM to all pages probably is something we'd want to do regardless.

I'd vote to reopen this pull

**Comment by jarstelfox** [comment]
Let's dev_block 🌵 this since we know it will have conflicts with https://github.com/iFixit/shopify-v2/pull/1121

As well, @sterlinghirsh should we still add gtm to all pages (it seems reasonable to me) even though it has no effect on piwik.

**Comment by danielcliu-ifixit** [comment]
Reclosing for now until we need GTM

**Pull request closed without merging**